### ISSUE TITLE
version checker fails if a tool is in detached HEAD state

### ISSUE THREAD

**Issue by seb-wahl** [esm-tools/esm_version_checker#14]
Although probably only developers will hit this error, I'll report it here.
I went back a few commits on a component to check whether an error I got is due to the latest update on a component. Hence the development install was in "detached head" state, which `esm_versions check` does not like:
```
Traceback (most recent call last):
  File "/home/shkifmsw/.local/bin/esm_versions", line 10, in <module>
    sys.exit(main())
  File "/sw/tools/anaconda3/2019.10/skl/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/sw/tools/anaconda3/2019.10/skl/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/sw/tools/anaconda3/2019.10/skl/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/sw/tools/anaconda3/2019.10/skl/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/sw/tools/anaconda3/2019.10/skl/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/shkifmsw/.local/lib/python3.7/site-packages/esm_version_checker/cli.py", line 107, in check
    message += f" (development install, on branch: {repo.active_branch.name}, describe={describe})"
  File "/home/shkifmsw/.local/lib/python3.7/site-packages/git/repo/base.py", line 705, in active_branch
    return self.head.reference
  File "/home/shkifmsw/.local/lib/python3.7/site-packages/git/refs/symbolic.py", line 272, in _get_reference
    raise TypeError("%s is a detached symbolic reference as it points to %r" % (self, sha))
TypeError: HEAD is a detached symbolic reference as it points to 'dd990fc402821fd35c3eecace9c4beaa022254d1'
```

**Comment by seb-wahl** [comment]
tested with the latest version (4.7.0)

**Comment by pgierz** [comment]
Fixed in 5.1.1
